backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt

```diff
@@ -26,23 +30,32 @@ internal class PostInlineLowering(val context: Context) : FileLoweringPass {
 
     private val symbols get() = context.ir.symbols
 
+    private val kTypeGenerator = KTypeGenerator(
+            context,
+            eraseTypeParameters = true // Mimic JVM BE behaviour until proper type parameter impl is ready.
+    )
+
     override fun lower(irFile: IrFile) {
         irFile.transformChildrenVoid(object : IrElementTransformerVoidWithContext() {
 
             override fun visitClassReference(expression: IrClassReference): IrExpression {
                 expression.transformChildrenVoid()
 
-                val builder = context.createIrBuilder(currentScope!!.scope.scopeOwnerSymbol,
-                        expression.startOffset, expression.endOffset)
+                val builder = createIrBuilder(expression)
 
-                return builder.irKClass(context, expression.symbol)
+                val symbol = expression.symbol
+                return if (symbol is IrClassSymbol) {
+                    builder.irKClass(context, symbol)
+                } else {
+                    // E.g. for `T::class` in a body of an inline function itself.
+                    builder.irCall(context.ir.symbols.ThrowNullPointerException.owner)
```

Comment:
NPE?

Reply:
Exactly.
Please note that this change doesn't actually affect current behaviour: it only moves handling this case here from `irKClass`.